pdl-compiler/src/ast.rs

```diff
@@ -154,10 +149,12 @@ pub enum FieldDesc {
 }
 
 #[derive(Debug, Serialize, Clone)]
-pub struct Field<A: Annotation> {
+pub struct Field {
     pub loc: SourceRange,
+    /// Unique identifier used to refer to the AST node in
+    /// compilation environments.
     #[serde(skip_serializing)]
-    pub annot: A::FieldAnnotation,
+    pub key: usize,
```

Comment:
nit: Could you create wrapper types to make sure that we don't mix Field and Decl keys ?

Reply:
Done